eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/implementation/EventDataUtil.java

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs.implementation;
+
+import com.azure.core.amqp.MessageConstant;
+import com.azure.core.implementation.util.ImplUtils;
+import com.azure.eventhubs.EventData;
+import org.apache.qpid.proton.Proton;
+import org.apache.qpid.proton.amqp.Binary;
+import org.apache.qpid.proton.amqp.Symbol;
+import org.apache.qpid.proton.amqp.messaging.AmqpValue;
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.qpid.proton.amqp.messaging.Data;
+import org.apache.qpid.proton.amqp.messaging.MessageAnnotations;
+import org.apache.qpid.proton.message.Message;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class for converting {@link EventData} to {@link Message}.
+ */
+public class EventDataUtil {
+    /**
+     * Maps the set of events given to a collection of AMQP messages.
+     */
+    public static List<Message> toAmqpMessage(String partitionKey, List<EventData> events) {
+        return events.stream().map(event -> toAmqpMessage(partitionKey, event)).collect(Collectors.toList());
+    }
+
+    /**
+     * Gets the serialized size of the AMQP message.
+     */
+    static int getDataSerializedSize(Message amqpMessage) {
+
+        if (amqpMessage == null) {
+            return 0;
+        }
+
+        int payloadSize = getPayloadSize(amqpMessage);
+
+        // EventData - accepts only PartitionKey - which is a String & stuffed into MessageAnnotation
+        final MessageAnnotations messageAnnotations = amqpMessage.getMessageAnnotations();
+        final ApplicationProperties applicationProperties = amqpMessage.getApplicationProperties();
+
+        int annotationsSize = 0;
+        int applicationPropertiesSize = 0;
+
+        if (messageAnnotations != null) {
+            for (Symbol value : messageAnnotations.getValue().keySet()) {
+                annotationsSize += sizeof(value);
+            }
+
+            for (Object value : messageAnnotations.getValue().values()) {
+                annotationsSize += sizeof(value);
+            }
```

Comment:
I haven't investigated what the messageAnnotations.getValue() method returns, but it seems to look a bit like a Map (with keySet() and values()). Have you considered instead calling `entrySet()` so you can do one O(n) operation rather than two O(n) operations?

Reply:
Good catch! I had copied this from Track 1 as-is.